crates/dojo-core/src/auth/systems.cairo

```diff
@@ -7,100 +7,157 @@ struct Route {
     resource_id: u250,
 }
 
+trait RouteTrait {
+    fn new(target_id: u250, role_id: u250, resource_id: u250) -> Route;
+}
+
+impl RouteImpl of RouteTrait {
+    fn new(target_id: u250, role_id: u250, resource_id: u250) -> Route {
+        Route { target_id, role_id, resource_id,  }
+    }
+}
+
 #[system]
 mod RouteAuth {
     use traits::Into;
-    use array::ArrayTrait;
 
-    use dojo_core::auth::components::Status;
-    use dojo_core::auth::components::Role;
+    use dojo_core::auth::components::AuthStatus;
+    use dojo_core::auth::components::AuthRole;
     use super::Route;
 
     use starknet::ContractAddress;
 
-    fn execute(ref routing: Array<Route>) {
-        _set_authorization_routing(ref routing, world_address);
-    }
-
-    fn _set_authorization_routing(ref routing: Array<Route>, world_address: starknet::ContractAddress) {
-        gas::withdraw_gas().expect('Out of gas');
-
-        if routing.is_empty() {
-            return ();
-        }
-
-        let r = routing.pop_front().unwrap();
-
-        let mut calldata = ArrayTrait::new(); 
-        serde::Serde::<Role>::serialize(@Role { id: r.role_id }, ref calldata);
-        IWorldDispatcher { contract_address: world_address }.set_entity(
-            dojo_core::string::ShortStringTrait::new('Role'), QueryTrait::new_from_id(r.target_id), 0_u8, calldata.span());
-
-        let mut calldata = ArrayTrait::new(); 
-        serde::Serde::<Status>::serialize(@Status { is_authorized: bool::True(()) }, ref calldata);
-        IWorldDispatcher { contract_address: world_address }.set_entity(
-            dojo_core::string::ShortStringTrait::new('Status'), (r.role_id, r.resource_id).into(), 0_u8, calldata.span());
+    fn execute(route: Route) {
+        // Set scoped role
+        commands::set_entity(
+            (route.target_id, route.resource_id).into(), (AuthRole { id: route.role_id })
+        );
 
-        _set_authorization_routing(ref routing, world_address);
+        // Set status
+        commands::set_entity(
+            (route.role_id, route.resource_id).into(),
+            (AuthStatus { is_authorized: bool::True(()) })
+        );
     }
 }
 
 #[system]
-mod Authorize {
+mod IsAccountAdmin {
     use traits::Into;
+    use starknet::get_tx_info;
+    use box::BoxTrait;
     use dojo_core::integer::u250;
-    use dojo_core::auth::components::Status;
-    use dojo_core::auth::components::Role;
+    use dojo_core::auth::components::AuthStatus;
+    use dojo_core::auth::components::AuthRole;
+
+    fn execute() -> bool {
+        // Get calling account contract address
+        let caller = get_tx_info().unbox().account_contract_address; // tx origin
```

Comment:
Hm one limitation this creates is the executor has to be an account. Maybe thats OK with starknet though, since even a DAO could be implemented as an account contract?